### PR TITLE
HZN-950: create sha1 files for Karaf repositories

### PR DESCRIPTION
**Pull request by RangerRick**
This is an alternate PR to https://github.com/OpenNMS/opennms/pull/1497 which does the sha1 generation in maven rather than generating after the fact in the debian/rpm packages.

* JIRA: http://issues.opennms.org/browse/HZN-950

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

File: container/karaf/pom.xml [hunk 1]
```diff
@@ -102,6 +102,31 @@
                 </executions>
             </plugin>
             <plugin>
+                <groupId>net.nicoulaj.maven.plugins</groupId>
+                <artifactId>checksum-maven-plugin</artifactId>
+                <version>1.5</version>
```
Comment (j-white) [inline container/karaf/pom.xml]: Can we set the version in the top-level pom instead in each of these?

File: opennms-full-assembly/pom.xml [hunk 2]
```diff
@@ -301,6 +301,33 @@
...
+                <include>**/*.jar</include>
+                <include>**/*.pom</include>
+                <include>**/*.war</include>
+                <include>**/*.xml</include>
+              </includes>
+            </fileSet>
+          </fileSets>
+        </configuration>
+      </plugin>
+      <!--
```
Comment (j-white) [inline opennms-full-assembly/pom.xml]: If we need to disable the `maven-clean-plugin` here I'd suggest adding a note as to why, or removing the block completely.

Comment (RangerRick) [inline opennms-full-assembly/pom.xml]: Oops, good catch... I had only turned it off while fixing things up so I could inspect the generated repo.  I'll push a fixed rebase.

File: container/karaf/pom.xml (hunk 1 above)
Comment (RangerRick) [inline container/karaf/pom.xml]: Alright, did that too.  New build should be happening shortly.

**Comment by j-white** [comment]
Thanks for the updates. Will merge once Bamboo says it's OK.

**Pull request merged**